core/src/ldml/ldml_processor.cpp

```diff
@@ -231,75 +233,89 @@ ldml_processor::process_event(
       }
       break;
     default:
+    // all other VKs
       {
-        // adapted from kmx_processor.cpp
-
-        /** a copy of the current/changed context, for transform use */
-        ldml::string_list ctxt;
-
-        // Construct a context buffer of all the KM_KBP_BT_CHAR items
-        // Extract the context into 'ctxt' for transforms to process
-        if (!!transforms) {
-          // if no transforms, no reason to do this extraction (ctxt will remain empty)
-          auto &cp = state->context();
-          // We're only interested in as much of the context as is a KM_KBP_BT_CHAR.
-          uint8_t last_type = KM_KBP_BT_UNKNOWN;
-          for (auto c = cp.rbegin(); c != cp.rend(); c++) {
-            last_type = c->type;
-            if (last_type != KM_KBP_BT_CHAR) {
-              // not a char, stop here
-              // TODO-LDML: markers?
-              break;
-            }
-            ctxt.emplace_front(1, c->character);
-            // extract UTF-32 to 1 or 2 UTF-16 chars in a string
-          }
-        }
-
         // Look up the key
         const std::u16string str = keys.lookup(vk, modifier_state);
+
         if (str.empty()) {
-          // not found, so pass the keystroke on to the Engine
+          // no key was found, so pass the keystroke on to the Engine
           state->actions().push_invalidate_context();
           state->actions().push_emit_keystroke();
           break; // ----- commit and exit
```

Comment:
Big question about what happens with transforms if no key matches. Do we have a set of default keys (i.e. QWERTY)? And if so, would they trigger transforms? Or are any undefined keys no-ops?

Reply:
fyi @rc-swag — _my understanding_ was that unknown keys get kicked back to the OS. That seems to be what the other processor does. this is how frame keys, enter, etc get handled also.  

I think that this breaks up the context then, doesn't it? ( a point of discontinuity)

Reply:
Not quite. Some frame keys cause a context invalidation. Frame keys and modifier keys etc then get kicked back to the OS, but any US English character-generating key gets added as an action:

https://github.com/keymanapp/keyman/blob/c0ec8ba76c623be503e9050ff289f7db094ccf07/core/src/kmx/kmx_processevent.cpp#L323-L327

So we don't lose context on those.

Reply:
I'm wondering if this could be a separate ticket? I'm definitely not clear on the implications here.

Reply:
hived off to https://github.com/keymanapp/keyman/issues/9451